cmd/talosctl/cmd/talos/edit.go

```diff
@@ -0,0 +1,170 @@
+		return WithClient(func(ctx context.Context, c *client.Client) error {
+			edit := editor.NewDefaultEditor([]string{
+				"TALOS_EDITOR",
+				"EDITOR",
+			})
+
+			resourceType := args[0]
+			if resourceType != config.Type {
+				return fmt.Errorf("only the config resource can be edited")
+			}
+
+			var lastError string
+
+			editFn := func(parentCtx context.Context, msg client.ResourceResponse) error {
+				if msg.Resource == nil {
+					return nil
+				}
+
+				if msg.Metadata.GetError() != "" {
+					fmt.Fprintf(os.Stderr, "%s: %s\n", msg.Metadata.GetHostname(), msg.Metadata.GetError())
+
+					return fmt.Errorf("aborted")
+				}
+
+				for {
+					var (
+						buf bytes.Buffer
+						w   io.Writer = &buf
+						id  string
+					)
+
+					metadata := msg.Resource.Metadata()
+
+					if metadata != nil {
+						id = metadata.ID()
+					}
+
+					if runtime.GOOS == "windows" {
+						w = crlf.NewCRLFWriter(w)
+					}
+
+					_, err := w.Write([]byte(
+						fmt.Sprintf(
+							"# Editing %s/%s at node %s\n", resourceType, id, msg.Metadata.GetHostname(),
+						),
+					))
+					if err != nil {
+						return err
+					}
+
+					if lastError != "" {
+						lines := strings.Split(lastError, "\n")
+
+						_, err = w.Write([]byte(
+							fmt.Sprintf("#\n# %s\n", strings.Join(lines, "\n# ")),
+						))
+						if err != nil {
+							return err
+						}
+					}
+
+					body, err := yaml.Marshal(msg.Resource.Spec())
+					if err != nil {
+						return err
+					}
+
+					_, err = w.Write(body)
+					if err != nil {
+						return err
+					}
+
+					edited, _, err := edit.LaunchTempFile(fmt.Sprintf("%s-%s-edit-", resourceType, id), ".yaml", &buf)
+					if err != nil {
+						return err
+					}
+
+					editedWithoutComments := bytes.TrimSpace(cmdutil.StripComments(edited))
+
+					if len(bytes.TrimSpace(editedWithoutComments)) == 0 {
+						fmt.Println("Apply was skipped: empty file.")
+
+						break
+					}
+
+					if bytes.Equal(
+						editedWithoutComments,
+						bytes.TrimSpace(cmdutil.StripComments(body)),
+					) {
+						fmt.Println("Apply was skipped: no changes detected.")
+
+						break
+					}
+
+					_, err = c.ApplyConfiguration(parentCtx, &machine.ApplyConfigurationRequest{
+						Data:      edited,
+						Immediate: editCmdFlags.immediate,
+					})
+					if err != nil {
+						lastError = err.Error()
+
+						continue
+					}
+
+					break
+				}
+
+				return nil
+			}
+
+			for _, node := range Nodes {
+				nodeCtx := client.WithNodes(ctx, node)
+				if err := helpers.ForEachResource(nodeCtx, c, editFn, editCmdFlags.namespace, args...); err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+	},
+}
+
+func init() {
+	editCmd.Flags().StringVar(&editCmdFlags.namespace, "namespace", "", "resource namespace (default is to use default namespace per resource)")
+	editCmd.Flags().BoolVar(&editCmdFlags.immediate, "immediate", true, "apply the change immediately (without a reboot)")
```

Comment:
can we please make it default to `false` to align with `apply config` so that we don't have a mismatch for now (we can later flip to be always immediate)?